haskell/private/dependencies.bzl

```diff
@@ -19,63 +20,115 @@ load(
 )
 load(":private/set.bzl", "set")
 
+def _cc_get_static_lib(lib_info):
```

Comment:
I have doubt about this strategy.  Why the order? Shouldn't the library artifact used taken based on the compilation context? If we build static, it should be a static library with the right pic status for example.

Reply:
> Shouldn't the library artifact used taken based on the compilation context?

For static vs. dynamic linking that's what `_cc_get_static_lib` and `_cc_get_dynamic_lib` do.

Before, Bazel had the `static_mode_params_for_executable` and `dynamic_mode_params_for_executable` attributes which would distinguish between library dependencies for static or dynamic linking mode. Now, they removed that distinction and instead provide direct access to the various variants of a library (`(pic_)static`, `dynamic`, `interface`). But, these can be `None` if a target does not provide a particular variant. E.g. for a `cc_library` in our test-suite I found that `static` is `None` and `pic_static` points to a `.a`. Similarly, I found that `interface` is `None` and `dynamic` points to a `.so`. These functions pick the most appropriate available variant. E.g. if linking statically, we'll prefer a static library. But, if none is available we'll pick a dynamic one. 

This PR makes no attempt to further exploit the new C++ API to e.g. distinguish pic static libraries from regular static libraries. This PR is only aiming at making rules_haskell work with the Bazel 0.22 C++ API again.

> If we build static, it should be a static library with the right pic status for example.

Where can the pic status for static linking be found? I don't see it available in `CcInfo` or `CcToolchainInfo`. Maybe I'm missing something?

Reply:
Great answer, thank you.